src/openapi/streaming/connection/transport/signalr-core-transport.ts

```diff
@@ -0,0 +1,623 @@
+import log from '../../../../log';
+import * as transportTypes from '../transportTypes';
+import * as constants from '../constants';
+import type {
+    ConnectionState,
+    TransportTypes,
+    StreamingMessage,
+} from '../../types';
+import type {
+    StreamingTransportOptions,
+    StreamingTransportInterface,
+    StreamingData,
+} from '../types';
+import type SignalR from '@microsoft/signalr';
+
+declare global {
+    interface Window {
+        signalrCore: typeof SignalR;
+    }
+}
+
+type Callback = (...args: any[]) => any;
+
+type RawCoreSignalRStreamingMessage = {
+    ReferenceId: string;
+    PayloadFormat: 1 | 2;
+    Payload: StreamingData;
+    MessageId: string;
+};
+
+const LOG_AREA = 'SignalrCoreTransport';
+const NOOP = () => {};
+
+// null at the end means stop trying and close the connection
+const RECONNECT_DELAYS = [0, 2000, 3000, 5000, 10000, null];
+
+const renewStatus = {
+    SUCCESS: 0,
+    INVALID_TOKEN: 1,
+    SESSION_NOT_FOUND: 2,
+};
+
+class SignalrCoreTransport implements StreamingTransportInterface {
+    baseUrl = '';
+    name = transportTypes.SIGNALR_CORE;
+    connection: SignalR.HubConnection | null = null;
+    authToken: string | null = null;
+    authExpiry: number | null = null;
+    contextId: string | null = null;
+    messageStream: SignalR.IStreamResult<any> | null = null;
+    lastMessageId: string | null = null;
+    hasStreamingStarted = false;
+    isDisconnecting = false;
+    hasTransportError = false;
+    state: ConnectionState = constants.CONNECTION_STATE_DISCONNECTED;
+
+    stateChangedCallback: (state: ConnectionState) => void = NOOP;
+    receivedCallback: (data: StreamingMessage) => void = NOOP;
+    errorCallback = NOOP;
+    unauthorizedCallback = NOOP;
+    setConnectionSlowCallback = NOOP;
+    transportFailCallback;
+
+    utf8Decoder!: TextDecoder;
+
+    constructor(baseUrl: string, transportFailCallback = NOOP) {
+        this.baseUrl = baseUrl;
+        // callbacks
+        this.transportFailCallback = transportFailCallback;
+
+        try {
+            this.utf8Decoder = new window.TextDecoder();
+        } catch (error) {
+            log.error(
+                LOG_AREA,
+                'Error occurred while initializing text decoder',
+                {
+                    error,
+                },
+            );
+
+            transportFailCallback();
+        }
+    }
+
+    static isSupported = function () {
+        return (
+            window.signalrCore &&
+            typeof window.signalrCore.HubConnectionBuilder === 'function' &&
+            typeof window.Uint8Array === 'function' &&
+            typeof window.TextDecoder === 'function' &&
+            // This check can be removed once signalr team resolves below issue
+            // https://github.com/dotnet/aspnetcore/issues/29424

```

Comment:
given the outcome here: https://github.com/dotnet/aspnetcore/issues/29424

maybe just remove the comment?